File: public/codes/bblsort/bblsort.py
# Python program for Bubble Sort

def bubbleSort(arr):
    n = len(arr)

    # Traverse through all array elements
    for i in range(n):

        # Last i elements are already in place
        for j in range(0, n-i-1):

            # traverse the array from 0 to n-i-1
            # Swap if the element found is greater
            # than the next element
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]


# Driver code to test above
arr = [100, 1, 10, -1, 0]

bubbleSort(arr)

print("The array after running bubble sort is : ", end="")
print(arr)

# Output :
# The array after running bubble sort is : [-1, 0, 1, 10, 100]

File: public/codes/bblsort/bblsort.c
// C program for Bubble Sort
#include <stdio.h>

// Function to sort an array using bubble sort
void bubbleSort(int arr[], int n)
{
    int i, j;
    for (i = 0; i < n - 1; i++)

        // Last i elements are already in place
        for (j = 0; j < n - i - 1; j++)
            //Swap if the element found is greater
            //than the next element
            if (arr[j] > arr[j + 1])
            {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
}

// A utility function to print an array of size n
void printArray(int arr[], int n)
{
    int i;
    printf("The array after running bubble sort is : [");
    for (i = 0; i < n - 1; i++)
        printf("%d, ", arr[i]);
    printf("%d]\n", arr[n - 1]);
}

// Driver code
int main()
{
    int arr[] = {100, 1, 10, -1, 0};
    int n = 5;

    bubbleSort(arr, n);
    printArray(arr, n);

    return 0;
}

// Output :
// The array after running bubble sort is : [-1, 0, 1, 10, 100]
